connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/outbound/ConnectorJobHandler.java

```diff
@@ -196,22 +205,27 @@ public void handle(final JobClient client, final ActivatedJob job) {
               job.getCustomHeaders().get(Keywords.RESULT_EXPRESSION_KEYWORD));
       result = new ConnectorResult.SuccessResult(response, responseVariables);
     } catch (ConnectorRetryException ex) {
+      Exception newException = new Exception(hideSecretsFromMessage(ex.getMessage(), secrets), ex);
       LOGGER.debug(
-          "ConnectorRetryException while processing job: {} for tenant: {}",
+          "ConnectorRetryException while processing job: {} for tenant: {}, error message: {}",
           job.getKey(),
           job.getTenantId(),
```

Comment:
Would be cool to have all the exception handling code in a single method outside of the already too big: 
```handle``` method.